Query hangs on invalid URI

### Describe the bug

While developping an app, I ended up forgetting about URI encoding the data and sent this:


http://server:8081/query?q=g\\+\\+&dataOnly=true

In the back end I saw this exception:
```java
//
Caused by: java.net.URISyntaxException: Illegal character in query at index 14: /query?q=g\\+\\+&dataOnly=true
//
```
but in the front end I ended up having a timeout without any error

### Expected behavior

The server doesn't hang and we have an error

### Actual behavior

The server hangs until the client timeouts

### How to Reproduce?

1. Create an http reactive-reasteasy 2.6.3.Final app
2. Send a malformed request: http://server:8081/query?q=g\\+\\+&dataOnly=true

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

hello, I just tried it with 2.7.3.Final, here is the stacktrace

```java
2022-03-04 14:30:10,285 WARN  [io.net.cha.AbstractChannelHandlerContext] (vert.x-eventloop-thread-3) An exception 'java.lang.NullPointerException' [enable DEBUG level for full stacktrace] was thrown by a user handler's exceptionCaught() method while handling the following exception:: java.lang.IllegalArgumentException: Illegal character in query at index 18: /utils/ping?data=g\\+\\
	at java.base/java.net.URI.create(URI.java:883)
	at io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer$ServerSpanNameExtractor.extract(OpenTelemetryVertxTracer.java:306)
	at io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer$ServerSpanNameExtractor.extract(OpenTelemetryVertxTracer.java:295)
	at io.opentelemetry.instrumentation.api.instrumenter.Instrumenter.start(Instrumenter.java:159)
	at io.opentelemetry.instrumentation.api.instrumenter.ServerInstrumenter.start(ServerInstrumenter.java:30)
	at io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer.receiveRequest(OpenTelemetryVertxTracer.java:96)
	at io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer.receiveRequest(OpenTelemetryVertxTracer.java:43)
	at io.vertx.core.http.impl.Http1xServerRequest.reportRequestBegin(Http1xServerRequest.java:580)
	at io.vertx.core.http.impl.Http1xServerRequest.handleBegin(Http1xServerRequest.java:153)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:143)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:156)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.websocketx.extensions.WebSocketServerExtensionHandler.channelRead(WebSocketServerExtensionHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:299)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.net.URISyntaxException: Illegal character in query at index 18: /utils/ping?data=g\\+\\
	at java.base/java.net.URI$Parser.fail(URI.java:2913)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3084)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3172)
	at java.base/java.net.URI$Parser.parse(URI.java:3125)
	at java.base/java.net.URI.<init>(URI.java:600)
	at java.base/java.net.URI.create(URI.java:881)

	... 41 more
```

--- a/extensions/opentelemetry/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/VertxClientOpenTelemetryTest.java
+++ b/extensions/opentelemetry/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/VertxClientOpenTelemetryTest.java
@@ -114,6 +114,18 @@ public class VertxClientOpenTelemetryTest {
         assertEquals(server.getParentSpanId(), client.getSpanId());
     }
 
+    @Test
+    void uri() throws Exception {
+        HttpResponse<Buffer> response = WebClient.create(vertx)
+                .get(uri.getPort(), uri.getHost(), "/hello?q=g\\+\\+&dataOnly=true")
+                .send()
+                .toCompletionStage().toCompletableFuture()
+                .get();
+
+        List<SpanData> spans = spanExporter.getFinishedSpanItems(2);
+        assertEquals(2, spans.size());
+    }
+
     @ApplicationScoped
     public static class HelloRouter {
         @Inject

--- a/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/OpenTelemetryVertxTracer.java
+++ b/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/OpenTelemetryVertxTracer.java
@@ -5,7 +5,6 @@ import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_
 import static io.quarkus.opentelemetry.runtime.OpenTelemetryConfig.INSTRUMENTATION_NAME;
 import static io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.setContextSafe;
 
-import java.net.URI;
 import java.util.List;
 import java.util.Map;
 import java.util.function.BiConsumer;
@@ -303,9 +302,8 @@ public class OpenTelemetryVertxTracer
         @Override
         public String extract(final HttpRequest httpRequest) {
             if (httpRequest instanceof HttpServerRequest) {
-                String path = URI.create(httpRequest.uri()).getPath();
-                if (path != null && path.length() > 1) {
-                    return path.substring(1);
+                if (httpRequest.uri() != null && httpRequest.uri().length() > 1) {
+                    return httpRequest.uri().substring(1);
                 } else {
                     return "HTTP " + httpRequest.method();
                 }

